www/blog/2023-01-08-tinyrpc-client.mdx

```diff
@@ -0,0 +1,210 @@
+---
+slug: tinyrpc-client
+title: Writing a tiny tRPC client
+author: Julius
+author_title: tRPC Contributor
+author_url: https://twitter.com/jullerino
+author_image_url: https://github.com/juliusmarminge.png
+---
+
+Ever wondered how tRPC works? Maybe you want to start contributing to the project but is freightened by the internals? The aim of this post is to familiarize you with the internals of tRPC by writing a minimal client that covers the big parts of how tRPC works.
+
+## Overview
+
+Let's assume we have simple tRPC router that looks like this:
+
+```ts
+const appRouter = t.router({
+  post: t.router({
+    byId: t.procedure
+      .input(z.string())
+      .query(({ input }) => posts.find((p) => p.id === input)),
+    create: t.procedure
+      .input(z.object({ title: z.string() }))
+      .mutation(({ input }) => {
+        const post = { id: uuid(), ...input };
+        posts.push(post);
+        return post;
+      }),
+  }),
+});
```

Comment:
Use the recommended syntax - (no `t.x`)

- `t.router` -> `router`
- `t.procedure` -> `publicProcedure` (I am annoyed with how long that is but let's be consistent)